internal/redfishwrapper/virtual_media.go

```diff
@@ -0,0 +1,74 @@
+package redfishwrapper
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	rf "github.com/stmcginnis/gofish/redfish"
+)
+
+// Set the boot device for the system.
+func (c *Client) SetVirtualMedia(ctx context.Context, kind string, mediaUrl string) (ok bool, err error) {
+	managers, err := c.Managers(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	var mediaKind rf.VirtualMediaType
+	switch kind {
+	case "CD":
+		mediaKind = rf.CDMediaType
+	case "Floppy":
+		mediaKind = rf.FloppyMediaType
+	case "USBStick":
+		mediaKind = rf.USBStickMediaType
+	case "DVD":
+		mediaKind = rf.DVDMediaType
+	default:
+		return false, errors.New("invalid media type")
+	}
+
+	for _, manager := range managers {
+		virtualMedia, err := manager.VirtualMedia()
+		if err != nil {
+			return false, err
+		}
+		for _, media := range virtualMedia {
+			if media.Inserted {
+				err = media.EjectMedia()
+				if err != nil {
+					return false, err
+				}
+			}
+		}
+	}
+
+	if mediaUrl != "" {
```

Comment:
Is `mediaUrl` a required parameter, if so, this method could return early if its not specified.

Reply:
Right now setting `mediaUrl = ""`  causes virtual media to be ejected. That seemed simpler than adding a whole separate method.

Reply:
oh right, now it makes sense - could you leave a comment on there unless I've missed it.